### DISCUSSION TITLE
Only redirect at checkout

### DISCUSSION THREAD

**Discussion by james-s-w-clark** [webdevnerdstuff/amazon-smile-redirect#100] [Ideas]
Current behaviour:
- Redirect most pages to Smile

Requested behaviour: 
- Only redirect checkout pages to Smile

There are some pros/cons with this change of behaviour:

- 👎  Users won't know they're using Smile until they check out (users may think the extension isn't working)
- 👍  Users won't see page unneccesary reloads, which isn't an ideal browisng experience when you regularly open many short-lived tabs

In theory I could retrain myself to Google search for "Smile" and select the correct hyperlink each time, but that's a fair bit of mental effort for most users.

---

This feature proposes adding a toggle to the extension's settings:

- `Only redirect at checkout` (default off)

**Comment by james-s-w-clark** [comment]
Interestingly, my browser is stuck in an infinite redirect loop for this item:
https://www.amazon.com/gp/product/B07VZLFW2M/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1&pldnSite=1
The AmazonSmile banner is at the top, but the URL never changes.

Even the url https://www.amazon.com/gp/product/B07VZLFW2M redirects me to https://www.amazon.com/gp/product/B07VZLFW2M?pldnSite=1, and has ininite redirects.

I don't know if that's a separate bug, but for now... this feature would have never let me see this issue.

**Reply by webdevnerdstuff** [reply]
I'm unable to replicate the issue (Chrome) with either of those links. Maybe try clearing your cache and restarting the browser to see if that resolves it.

**Reply by james-s-w-clark** [reply]
@webdevnerdstuff I assume you're native to amazon.com - how about https://www.amazon.de/kindle-cover-Elektronik-Foto/s?k=kindle+cover&rh=n%3A562066&pldnSite=1 ? 

I tried amazon.es/.it, and they had no redirects (maybe Smile isn't there, no worries)

amazon.de does seem to have smile, but I didn't choose a charity there (and I don't have an account there). It's like the extension wants to take my me to a Smile version of the current page, but first the user should sign up to Smile and handle the preamble at e.g. https://smile.amazon.de/charity?orig=%2F

---

It might be uncommmon (but possible?) for someone to use Smile on different Amazon countries' websites.

This might be a separate bug, if amazon.de can reproduce it for you.

**Comment by webdevnerdstuff** [comment]
I have now added this as a new option. I still have some more testing to do with it, but so far it's working as expected (at least in one browser so far). It'll take some time before the next release as I'm still in the middle of rebuilding/optimizing things. I also need to add in the translations for all of the languages for this new menu item, which tends to be a bit of a tedious process.

But the good news is that it looks like it's working good so far, and will launch with the next release.

**Reply by james-s-w-clark** [reply]
Amazing, thank you!

**Comment by webdevnerdstuff** [comment]
Updated extension has been submitted to all browsers and Version 3.0.0 should be live as soon as they have been approved.

Pull Request: https://github.com/webdevnerdstuff/amazon-smile-redirect/pull/104

**Reply by james-s-w-clark** [reply]
Thank you @webdevnerdstuff !

**Reply by webdevnerdstuff** [reply]
If you haven't already, I'd appreciate if you'd drop a positive review for the extension if you like it. :)

**Reply by james-s-w-clark** [reply]
@webdevnerdstuff Done!

**Comment by james-s-w-clark** [comment]
Can you close discussions? I think this one is done :)

**Comment by webdevnerdstuff** [comment]
They can only be deleted, not closed. It's all good to keep it up here incase someone doesn't know about the feature.

**Reply by james-s-w-clark** [reply]
Thanks for explaining :) sounds good!
